src/math/tests/BarycenterTest.cpp

```diff
@@ -237,6 +238,183 @@ BOOST_AUTO_TEST_CASE(BarycenterTriangle2D)
   }
 }
 
+struct TetrahedronFixture {
+  Eigen::Vector3d a{0.0, 0.0, 0.0};
+  Eigen::Vector3d b{0.0, 1.0, 0.0};
+  Eigen::Vector3d c{1.0, 0.0, 0.0};
+  Eigen::Vector3d d{0.0, 0.0, 1.0};
+
+  Eigen::Vector3d center{0.25, 0.25, 0.25};
+  Eigen::Vector4d center_coords{0.25, 0.25, 0.25, 0.25};
+};
+
+struct FlippedTetrahedronFixture {
+  Eigen::Vector3d a{0.0, 0.0, 0.0};
+  Eigen::Vector3d b{0.0, 1.0, 0.0};
+  Eigen::Vector3d c{1.0, 0.0, 0.0};
+  Eigen::Vector3d d{0.0, 0.0, -1.0};
+
+  Eigen::Vector3d center{0.25, 0.25, -0.25};
+  Eigen::Vector4d center_coords{0.25, 0.25, 0.25, 0.25};
+};
+
+struct AlmostDegenerateTetrahedronFixture {
+  Eigen::Vector3d a{0.0, 0.0, 0.0};
+  Eigen::Vector3d b{0.0, 1.0, 0.0};
+  Eigen::Vector3d c{1.0, 0.0, 0.0};
+  Eigen::Vector3d d{0.0, 0.0, 0.001};
+
+  Eigen::Vector3d center{0.25, 0.25, 0.00025};
+  Eigen::Vector4d center_coords{0.25, 0.25, 0.25, 0.25};
+};
+
+struct FunnyTetrahedronFixture {
+  Eigen::Vector3d a{0.0, 0.0, 0.0};
+  Eigen::Vector3d b{0.0, 1.0, 0.0};
+  Eigen::Vector3d c{1.0, 0.0, 0.0};
+  Eigen::Vector3d d{11.0, 11.0, 1.0};
+
+  Eigen::Vector3d center{3.0, 3.0, 0.25};
+};
+
+typedef boost::mpl::vector<TetrahedronFixture, FlippedTetrahedronFixture, AlmostDegenerateTetrahedronFixture, FunnyTetrahedronFixture> TetrahedraFixtures;
+
+BOOST_FIXTURE_TEST_CASE_TEMPLATE(BarycenterTetrahedronVertices, T, TetrahedraFixtures, T)
+{
+  PRECICE_TEST(1_rank);
+  using Eigen::Vector3d;
+  using Eigen::Vector4d;
+  using precice::testing::equals;
+
+  // Alias for fixture data
+  const auto &a = T::a;
+  const auto &b = T::b;
+  const auto &c = T::c;
+  const auto &d = T::d;
```

Comment:
The point of the test fixture is that you can directly use members of the fixture inside the test. So this is obsolete.